### ISSUE TITLE
Rearranging the VIRTUALKEYBOARDS files and conventions

### ISSUE THREAD

**Issue by rmkaplan** [Interlisp/medley#1267]
VIRTUALKEYBOARDS was written at a time when there was a close correlation between machine types and keyboard types (if you had a Dorado, then you had a Dorado keyboard).  The recent discussion of KEYBOARDTYPE points out that this is no longer the case.

Separately, the files that are all connected to VIRTUALKEYBOARDS (KEYBOARDCONFIGS, DORADOKEYBOARDS, MAIKOKEYBOARDS...) are spread around library/ in an unrelated way. That was the case with other multi-file applications (Lafite, Tmax...) before we implemented the library/xxx/ convention where FINDFILE and the various system tools would look for library/xxx/xxx-yyy files as if they were at the higher library/ level.  This has made it possible to add some useful organization to library/ and other directories.

Finally, the keyboard datafiles (DORADOKEYBOARDS...) currently have to be read and written in a special way, because the conventions for them were set up before readtables, packages, etc. were generalized--in particular, they don't have the proper self-describing DEFINE-FILE-INFO's .

I am working to address these issues in the following ways:

1.  Remove references to MACHINETYPE from VIRTUALKEYBOARDS, and replace them with calls to KEYBOARDTYPE (which itself has to be fixed up--see issue #1265).  I will rename the keyboards that used to be named in particular for MACHINETYPE MAIKO to now be named for KEYBOARDTYPE X.  We can then add other keyboard types (SDL?) if they diverge from X even when running Maiko.  To that end, I will also rename the file MAIKOKEYBOARDS to XKEYBOARDS.

2. Create library/virtualkeyboards, and move all the relevant files (VIRTUALKEYBOARDS, KEYBOARDCONFIGS, DORADOKEYBOARDS...) into that.  The data files don't have code, so I don't think it matters that they are not renamed as e.g. VIRTUALKEYBOARDS-DORADOKEYBOARDS for the benefit of FINDCALLERS.  And loading is done internal to VIRTUALKEYBOARDS, so it knows where they are.

(The only glitch in this scheme is KEYBOARDEDITOR, which is a code file for creating/editing keyboards for different languages in different configurations.  It is described by that name in the Library documentation, so it would be awkward to change that to VIRTUALKEYBOARDS-KEYBOARDEDITOR and move it down out of library/.  So for now I propose to leave it where it is.)

3.  Convert the existing keyboard files from their idiosyncratic format to standard managed files that come with a proper DEFINE-FILE-INFO.  The files can then be LOADed in the ordinary way.  The loaded keyboards will be stored on a new variable VKBD.LOADED-KEYBOARDS, an alist indexed by keyboardtype.  So for example, XKEYBOARDSCOMS would just be specified as ((ALISTS (VKBD.LOADED-KEYBOARDS X)).  The keyboards of a particular type can then be activated by setting the legacy variable VKDB.KNOWN-KEYBOARDS to the entry for that type in VKBD.LOADED-KEYBOARDS. (Presumably in any one instantiation only the file for the local keyboard would actually be loaded in.)

I'll note, by the way, that my initials are all over this stuff, even though I don't remember much about it.   I think LFG was the main driver/user for this capability.

Comments and suggestions?

**Comment by rmkaplan** [comment]
A further comment/issue about KEYBOARDEDITOR:  current (master/release) version seems out of step with the master/release) version of VIRTUALKEYBOARDS.  It calls at least one function with the wrong arguments and it calls another function (VKBD.ASSIGNABLE-KEYP) that doesn't exist.  

Otherwise, I have more or less completed the revisions outlined above.

**Comment by rmkaplan** [comment]
I tried loading the VIRTUALKEYBOARDS in this PR into the full.sysout in the just-released package, with -nogreet to avoid anything in my init.  Without greeting it was a little tricky to get the directories set right (they start as home/runner/work), but after a little fiddling, the load went through without incident.  I did not see the DEFAULT problem that you observed.  The keyboards worked as expected.

I also made a loadup myself in the current master, ran it after switching to his PR and loaded its VIRTUALKEYBOARDS, and that also seemed to act in the right way.

So I can't replicate the DEFAULT problem
